sparkctl/cmd/create.go

```diff
@@ -267,28 +272,120 @@ func isContainerLocalFile(file string) (bool, error) {
 	return false, nil
 }
 
+type Blob interface {
+	// TODO: With go-cloud supporting setting ACLs, remove implementations of interface
+	setPublicACL(ctx context.Context, bucket string, filePath string) error
+}
+
+type BlobHandler struct {
+	ctx    context.Context
+	bucket *blob.Bucket
+	blob   Blob
+}
+
+func (bh BlobHandler) uploadToBucket(
+	uploadScheme string,
+	uploadBucket string,
+	uploadPath string,
+	localFilePath string) (string, error) {
+	fileName := filepath.Base(localFilePath)
+	uploadFilePath := filepath.Join(uploadPath, fileName)
+	// Check if exists by trying to fetch metadata
+	reader, err := bh.bucket.NewRangeReader(bh.ctx, uploadFilePath, 0, 0)
+	defer reader.Close()
```

Comment:
It looks like you don't even need to use `defer` here since it's closed immediately after being created.

Reply:
Not sure if I understand.

Reply:
You don't need to use `defer` as `reader` is not used anywhere below, it's closed right after it's created.